core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -526,6 +537,19 @@ private[spark] class ExecutorAllocationManager(
       allocationManager.synchronized {
         stageIdToNumTasks(stageId) = numTasks
         allocationManager.onSchedulerBacklogged()
+
+        var numTasksPending = 0
```

Comment:
there may has chance localities is empty, so using taskLocalityPreferences.size() to calculating the number of locality preferred pending task is not accurate.